discovery-provider/src/tasks/user_replica_set.py

```diff
@@ -0,0 +1,229 @@
+                        update_task,
+                        primary,
+                        secondaries,
+                        redis
+                    )
+                    user_record.creator_node_endpoint = creator_node_endpoint_str 
+                    user_replica_set_events_lookup[user_id]["user"] = user_record
+                    user_replica_set_events_lookup[user_id]["events"].append(event_type)
+                elif event_type == user_replica_set_manager_event_types_lookup['add_or_update_content_node']:
+                    # TODO: Handle indexing creator node fields separately
+                    # PROCESSING THIS IS PENDING BELOW PR:
+                    # https://github.com/AudiusProject/audius-protocol/pull/1163/files
+                    # This is in order to enable bootstrap node events to be indexed outside of the constructor
+                    #   without a significant modification to the indexing flow.
+                    logger.error('add_or_update_content_node EVENT FOUND')
+                    logger.error(f'{event_type}')
+                    logger.error(args)
+                    cnode_record = parse_poa_cnode_record(
+                        self,
+                        update_task,
+                        session,
+                        entry,
+                        cnode_events_lookup[cnode_id]["content_node"]
+                    )
+                    if cnode_record is not None:
+                        cnode_events_lookup[cnode_id]["content_node"] = cnode_record
+                        cnode_events_lookup[cnode_id]["events"].append(event_type)
+            num_total_changes += len(user_events_tx)
+
+    # for each record in user_replica_set_events_lookup, invalidate the old record and add the new record
+    # we do this after all processing has completed so the user record is atomic by block, not tx
+    for user_id, value_obj in user_replica_set_events_lookup.items():
+        logger.info(f"user_replica_set.py | Replica Set Processing Adding {value_obj['user']}")
+        invalidate_old_user(session, user_id)
+        session.add(value_obj["user"])
+
+    for content_node_id, value_obj in cnode_events_lookup.items():
+        logger.info(f"user_replica_set.py | Content Node Processing Adding {value_obj['content_node']}")
+        invalidate_old_cnode_record(session, content_node_id)
+        session.add(value_obj["content_node"])
+
+    return num_total_changes, user_ids
+
+# TODO: Enable a cache from index_network_peers
+def get_endpoint_string_from_sp_ids(
+    self,
+    update_task,
+    primary,
+    secondaries,
+    redis
+):
+    shared_config = update_task.shared_config
+    eth_web3 = update_task.eth_web3
+    eth_registry_address = update_task.eth_web3.toChecksumAddress(
+        shared_config["eth_contracts"]["registry"]
+    )
+    logger.error(f"USER REPLICA SET PY: ETH REGISTRY {eth_registry_address}")
+    eth_registry_instance = eth_web3.eth.contract(
```

Comment:
TODO - Enable a cache from index_network_peers - this would mean both storing values in `index_network_peers.py` retrieving them here and falling back if necessary